chore: Improve PR checklist based on retro feedback

evanaronson opened the pull request:
# Description

Updated the pull request template to include new task options and modified checklist items for clarity and removing jira link since Linear is smarter than Jira.

jamesej reviewed (approved):
All looks good thanks

evanaronson committed:
Revise pull request template

Updated the pull request template to include new task options and modified checklist items for clarity.

Signed-off-by: Evan Aronson <93671071+evanaronson@users.noreply.github.com>

thekidnamedkd reviewed (approved):
LGTM ✨
